core/templates/domain/collection/collection.model.ts

```diff
@@ -35,32 +35,34 @@ export interface CollectionBackendDict {
   'id': string;
   'title': string;
   'objective': string;
-  'language_code': string;
-  'tags': string[];
-  'schema_version': number;
+  'language_code': string | null;
+  'tags': string[] | null;
+  'schema_version': number | null;
   'playthrough_dict': CollectionPlaythroughBackendDict;
   'category': string;
   'version': number;
   'nodes': CollectionNodeBackendDict[];
 }
 
 export class Collection {
-  id: string;
-  title: string;
-  objective: string;
-  languageCode: string;
-  tags: string[];
+  id: string | null;
+  title: string | null;
+  objective: string | null;
+  languageCode: string | null;
+  tags: string[] | null;
```

Comment:
Instead of allowing null on all values maybe we should not create the objects if some crucial values are not defined?

Reply:
Can you check the places where we create an empty collection and see if it can be refactored there?

Reply:
Ok. Sure

Reply:
core/templates/domain/collection/collection-rights.model.ts
```
static createEmptyCollectionRights(): CollectionRights {
    return new CollectionRights({
      owner_names: [],
      collection_id: null,
      can_edit: null,
      can_unpublish: null,
      is_private: null
    });
  }
```

core/templates/domain/collection/collection.model.ts

```
static createEmptyCollection(): Collection {
    let emptyCollectionPlaythrough = CollectionPlaythrough.create(null, []);
    return new Collection(
      null, null, null, null, null, emptyCollectionPlaythrough,
      null, null, null, []);
  }
```

Reply:
I meant places where we use `createEmptyCollection`.

Reply:
Ok

Reply:
`core/templates/pages/collection-editor-page/services/collection-editor-state.service.ts`

`core/templates/pages/collection-editor-page/services/collection-linearizer.service.spec.ts`
`core/templates/domain/collection/CollectionObjectFactorySpec.ts`

Reply:
And can it be refactored there so that we do not need to create empty collection?

Reply:
I am looking into the code